src/salad_cloud_transcription_sdk/services/transcription.py

```
@@ -0,0 +1,289 @@
+            organization_name=organization_name,
+            inference_endpoint_name=inference_endpoint_name,
+            inference_endpoint_job_id=job_id,
+        )
+
+    def process_webhook_request(
+        self,
+        payload: Any,
+        signing_secret: str,
+        webhook_id: str,
+        webhook_timestamp: str,
+        webhook_signature: str,
+    ) -> TranscriptionWebhookPayload:
+        """Process a webhook request from Salad Cloud Transcription service.
+
+        :param payload: The webhook request payload (string or bytes)
+        :type payload: Any
+        :param signing_secret: The secret used for verifying the webhook signature
+        :type signing_secret: str
+        :param webhook_id: The webhook ID from the request header
+        :type webhook_id: str
+        :param webhook_timestamp: The timestamp from the request header
+        :type webhook_timestamp: str
+        :param webhook_signature: The signature from the request header
+        :type webhook_signature: str
+
+        :raises WebhookVerificationError: If signature validation fails
+
+        :return: The processed job result
+        :rtype: InferenceEndpointJob
+        """
+        # Create headers dictionary for verification
+        headers = {
+            "webhook-id": webhook_id,
+            "webhook-timestamp": webhook_timestamp,
+            "webhook-signature": webhook_signature,
+        }
+
+        # Initialize webhook validator with the signing secret
+        webhook = Webhook(signing_secret)
+
+        # Verify the payload signature
+        # This will raise WebhookVerificationError if validation fails
+        if webhook.verify(payload, headers):
+            deserialized_payload = TranscriptionWebhookPayload.from_json(payload)
+            deserialized_payload.data = self._convert_job_output(
+                deserialized_payload.data
+            )
+            return deserialized_payload
+
+        return None
```

the method returns None if verification fails. Should we raise an exception instead?

---

That's true. I'll update.